src/main/java/org/otherband/Utils.java

```diff
@@ -11,4 +15,11 @@ private Utils() {
     public static void validateNotBlank(String field, String fieldName) {
         if (isBlank(field)) throw UserInputException.formatted("[%s] cannot be blank", fieldName);
     }
+
+    public static String buildLetterVerificationLink(VerificationLinkEmailRequest verificationLinkEmailRequest) {
+        return linkTo(methodOn(RecommendationLetterController.class)
+                .verify(verificationLinkEmailRequest.getLetterId(),
+                        verificationLinkEmailRequest.getTokenId(),
+                        verificationLinkEmailRequest.getSecretToken())).withSelfRel().toString();
+    }
```

Comment:
Can we move this back to EmailServiceSimulator.java as this method is not for general usage and is strictly related to the EmailService

Reply:
Yeah but we have multiple implementations of EmailService interface so we need it in multiple places. I was thinking of making it part of the VerificationLinkEmailRequest class but @yazan-abu-obaideh said we shouldn't add methods to DTO classes so we decided to put it in Utils

Reply:
Makes absolute sense 💯